Lib/test/librptest/core.py

```diff
@@ -0,0 +1,76 @@
+'''
+Core Functions Logic to manage and evaluate test case annotations.
+'''
+
+import types
+
+
+
+substitutions = {}
+marked_test_cases = {}
+
+### Adds a function as a substitute of the original function fct.
+def add_subst(fct, subst, reason=None):
+    if fct==None:
+        return
+    
+    if fct in substitutions:
+        substitutions[fct].add((subst, reason))
+    else:
+        substitutions[fct]=set([(subst, reason)])
```

Comment:
Is there a reason not to use `defaultdict(set)`?

Reply:
Yes, because in RustPython is crashes. I will raise a ticket for that...